cpp/independent_set/IndependentSet.cpp

```diff
@@ -0,0 +1,318 @@
+            visitedNodes.push_back(n2);
+        });
+    }
+
+    std::vector<bool> visitedOnGraph(graph->numberOfNodes());
+    visitedOnGraph[v] = true;
+    for (auto x : neighbors) {
+        visitedOnGraph[x] = true;
+    }
+
+    std::vector<NetworKit::node> neighbors2;
+    for (auto x : visitedNodes) {
+        if (!visitedOnGraph[x]) {
+            visitedOnGraph[x] = true;
+            neighbors2.push_back(x);
+        }
+    }
+    return neighbors2;
+}
+
+IndependentSet::EdgeSet IndependentSet::getConnectedEdges(std::vector<NetworKit::node>& nodes) const {
+    EdgeSet connectedEdges(edgeComparator);
+    for (auto x : nodes) {
+        graph->forEdgesOf(x, [&](NetworKit::node u, NetworKit::node v) {
+            connectedEdges.insert(NetworKit::Edge(u, v, true));
+        });
+    }
+    return connectedEdges;
+}
+
+IndependentSet::EdgeSet IndependentSet::getInducedEdges(std::vector<NetworKit::node>& nodes) const {
+    EdgeSet connectedEdges(edgeComparator); 
+    std::set<NetworKit::node> nodeSet(nodes.begin(), nodes.end());
+
+    for (auto u : nodes) {
+        graph->forEdgesOf(u, [&](NetworKit::node u, NetworKit::node v) {
+            if (nodeSet.contains(v)) {
+                connectedEdges.insert(NetworKit::Edge(u, v, true));
+            }
+        });
+    }
+    return connectedEdges;
+}
+
+IndependentSet::EdgeSet IndependentSet::getAllEdges() const {
+    EdgeSet edges(edgeComparator);
+    graph->forEdges([&](NetworKit::node u, NetworKit::node v) {
+        edges.insert(NetworKit::Edge(u, v, true));
+    });
+    return edges;
+}
+
+std::vector<NetworKit::node> IndependentSet::getAllNodes() const {
+    std::vector<NetworKit::node> nodes;
+    graph->forNodes([&](NetworKit::node v) {
+        nodes.push_back(v);
+    });
+    return nodes;
+}
+
+std::vector<NetworKit::node> IndependentSet::getMirrors(NetworKit::node v) const {
+    std::vector<NetworKit::node> mirrors;
+    std::vector<NetworKit::node> neighbors2 = getNeighbors2(v);
+    std::vector<NetworKit::node> neighborsV = getNeighbors(v);
+
+    for (auto w : neighbors2) {
+        std::vector<NetworKit::node> neighborsW = getNeighbors(w);
+        std::set<NetworKit::node> potentialClique(neighborsV.begin(), neighborsV.end());
+        for (auto node : neighborsW) {
+            potentialClique.erase(node);
+        }
+
+        bool clique = true;
+        for (auto a : potentialClique) {
+            for (auto b : potentialClique) {
+                if (a != b && !graph->hasEdge(a, b)) {
+                    clique = false;
+                }
+            }
+        }
+        if (clique) {
+            mirrors.push_back(w);
+        }
+    }
+    return mirrors;
+}
+
+void IndependentSet::dfs(NetworKit::node v, std::vector<bool>& visited) {
+    if (visited[v]) {
+        return;
+    }
+    visited[v] = true;
+    std::vector<NetworKit::node> neighbors = getNeighbors(v);
+    for (auto n : neighbors) {
+       dfs(n, visited);
+    }
+}
```

Comment:
Why can't you use default NetworKit DFS? There is some, and I guess you can pass a lambda there